### DISCUSSION TITLE
Suppress Output

### DISCUSSION THREAD

**Discussion by Leighton365** [pnp/powershell#2626] [Q&A]
Hi all,
I was wondering, is it possible to suppress output from PnP commands.
I have tried | out-null and a number of other options [void] etc, but nothing appears to work.
My script has a menu system and as the user runs each option, I'd like it to show that section is in progress, rather than all of the command output.  I am using PnP-TraceLog to capture information/errors and determine whether there was an error.
Just trying to make everything as neat as possible.
Thanks in advance!

Leighton

**Comment by AndersRask** [comment]
Can you please give code samples?

Out-null should do it

søn. 11. dec. 2022 14.20 skrev Leighton365 ***@***.***>:

**Reply by Leighton365** [reply]
Hi AndersRask,
Thanks for the quick reply.

The one line i'd really like to suppress output on is:
 $null = Invoke-PnPSiteTemplate -InputInstance $tmpLists -Connection $global:newsite -ErrorAction Silentlycontinue | Out-Null

It really screws up the terminal as it runs. It's in a foreach loop.

**Comment by AndersRask** [comment]
Having an assignment when using out-null does not make sense ;)

In what way does it interfere? It should not send output to any stream
(output, warning, verbose)

søn. 11. dec. 2022 15.30 skrev Leighton365 ***@***.***>:

**Reply by Leighton365** [reply]
It didn't have an assignment at first, but I started throwing everything at it!
It doesn't seem to make a difference what I do, I still get the command output.

**Reply by AndersRask** [reply]
Can you be a bit more specific on what fails?

Examples :)

**Reply by AndersRask** [reply]
The only thing I can think of could interfer depending on how you made your script menu is the progress output to screen.

Try and disable that by setting progress preference to SilentlyContinue:

`$progressPreference = 0`

**Reply by Leighton365** [reply]
Here's an example of the output I get even piping to Out-Null
![Untitled](https://user-images.githubusercontent.com/86303200/206911501-588ae8b6-3b9e-4c8f-8678-486f28922f13.png)

I'm sure it's not my menu, as the command always outputs to screen, even when I run it on it's own.

**Comment by AndersRask** [comment]
That is not output it is just process indicator.

Setting process default as specified in my earlier message should disable
this

søn. 11. dec. 2022 16.09 skrev Leighton365 ***@***.***>:

**Reply by Leighton365** [reply]
Aha!
That's fixed it.  Thank you so much!
